modules/activity_tracker/main.tf

```diff
@@ -38,3 +38,116 @@ resource "logdna_archive" "archive_config" {
     resourceinstanceid = var.cos_instance_id
   }
 }
+
+########################################################################
+# Activity Tracker Event Routing
+#########################################################################
+
+# COS targets
+resource "ibm_atracker_target" "atracker_cos_targets" {
+  for_each = var.cos_targets
+  cos_endpoint {
+    endpoint   = each.value.cos_endpoint.endpoint
+    bucket     = each.value.cos_endpoint.bucket_name
+    target_crn = each.value.cos_endpoint.target_crn
+    api_key    = each.value.cos_endpoint.api_key
+  }
+  name        = each.key
```

Comment:
Is name important here? because I'm not sure what the value of `each.key` is going to be here, and it won't use our normal prefix, so will not be unique.

Reply:
yes, name is required field and it will be key of element in a map

```
  logdna_targets = {
    "${var.prefix}-logdna-target-1" = {
      logdna_endpoint = {
        target_crn    = module.logdna_1.crn
        ingestion_key = module.logdna_1.ingestion_key
      }
      target_region = local.logdna_target_region
    }

    "${var.prefix}-logdna-target-2" = {
      logdna_endpoint = {
        target_crn    = module.logdna_2.crn
        ingestion_key = module.logdna_2.ingestion_key
      }
      target_region = local.logdna_target_region
    }
  }
```

Reply:
yes, name (target_name) is required field
```
  logdna_targets = {
    "${var.prefix}-logdna-target-1" = {
      instance_id   = module.logdna_1.crn
      ingestion_key = module.logdna_1.ingestion_key
      target_region = local.logdna_target_region
    }

    "${var.prefix}-logdna-target-2" = {
      instance_id   = module.logdna_2.crn
      ingestion_key = module.logdna_2.ingestion_key
      target_region = local.logdna_target_region
    }
  }
  ```
  key of each element in a map will be used a name